flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/io/StreamMultipleInputProcessor.java

```diff
@@ -55,17 +77,24 @@ public StreamMultipleInputProcessor(
                 || inputSelectionHandler.areAllInputsFinished()) {
             return AVAILABLE;
         }
-        final CompletableFuture<?> anyInputAvailable = new CompletableFuture<>();
         for (int i = 0; i < inputProcessors.length; i++) {
             if (!inputSelectionHandler.isInputFinished(i)
-                    && inputSelectionHandler.isInputSelected(i)) {
-                assertNoException(
-                        inputProcessors[i]
-                                .getAvailableFuture()
-                                .thenRun(() -> anyInputAvailable.complete(null)));
+                    && inputSelectionHandler.isInputSelected(i)
+                    && inputProcessors[i].getAvailableFuture() == AVAILABLE) {
+                return AVAILABLE;
```

Comment:
Isn't this a duplicate of `inputSelectionHandler.isAnyInputAvailable()` check from the `if` statement above (L54:56 in the original code)?

Reply:
the for loop through L80 to L86 is try to do some short cut if any of the inputProcessor's avaiableFuture is **AVAILABLE**, avoid the following new instance creation.
The `if` condition is identical with the original.

Reply:
But
1. in what scenario is this extra short cut doing something more compared to the old check?
2. if there is such scenario, have you tested that it's worth the added complexity?